src/punycode.cpp

```diff
@@ -0,0 +1,335 @@
+
+                // let i = i + digit * w, fail on overflow
+                if (digit > ((MAX_PUNYCODE_UINT - i) / w))
+                {
+                    throw std::invalid_argument("Overflow on i.");
+                }
+                i += digit * w;
+
+                // let t = tmin if k <= bias {+ tmin}, or
+                //         tmax if k >= bias + tmax, or k - bias otherwise
+                punycode_uint t = k <= bias ? TMIN :
+                                  k >= bias + TMAX ? TMAX : k - bias;
+
+                // if digit < t then break
+                if (digit < t)
+                {
+                    break;
+                }
+                
+                // let w = w * (base - t), fail on overflow
+                if (w > (MAX_PUNYCODE_UINT / (BASE - t)))
+                {
+                    // I believe this line is unreachable without first overflowing i.
+                    // Since 'i' is updated above as i += digit * w, and w is updated as
+                    // w = w * (BASE - t), we should like to keep (BASE - t) > digit to
+                    // give 'w' a chance to overflow first. To keep t minimized, we must
+                    // have 'bias' maximized. `bias` is driven by the 'adapt' function
+                    // below.
+                    //
+                    // The value returned by 'adapt' increases with the input delta, and
+                    // decreases with the input size. The delta is a function of the input
+                    // size as well, on the order of (delta_n * input size), and
+                    // legitimate delta_n values are limited to 0x10FFFF (the maximum
+                    // unicode codepoint). Even setting that aside, the maximum value that
+                    // adapt() can return is adapt(2 ** 32 - 1, 1, false) = 204.
+                    //
+                    // Using this bias, we could use the input (HERE) to get iterations:
+                    //
+                    //     digit = b = 1, i = 2, k = 36, t = 1, w = 35
+                    //     digit = b = 1, i = 37, k = 72, t = 1, w = 1225
+                    //     digit = b = 1, i = 1262, k = 108, t = 1, w = 42875
+                    //     digit = b = 1, i = 44137, k = 144, t = 1, w = 1500625
+                    //     digit = b = 1, i = 1544762, k = 180, t = 1, w = 52521875
+                    //
+                    // At this point, t now becomes TMAX (26) becuase k exceeds the bias
+                    // (since the maximum bias is 204). As such, the minimum continuation
+                    // value is 26:
+                    //
+                    //     digit = 0 = 26, i = 1367113512, k = 216, t = 26, w = 525218750
+                    //
+                    // However, the next iteration now overflows i before we can get to
+                    // the w update.
+                    throw std::invalid_argument("Overflow on w."); // LCOV_EXCL_LINE
```

Comment:
This is the single, solitary line for which I cannot get coverage, but I can't get it to honor the `LCOV_EXCL_LINE` comment. Any suggestions?